src/back/spv/writer.rs

```diff
@@ -1768,19 +2127,77 @@ impl Writer {
             }
             crate::Expression::LocalVariable(variable) => function.variables[&variable].id,
             crate::Expression::Load { pointer } => {
-                let (pointer_id, _) =
-                    self.write_expression_pointer(ir_function, fun_info, pointer, block, function)?;
+                match self.write_expression_pointer(
+                    ir_module,
+                    ir_function,
+                    fun_info,
+                    pointer,
+                    block,
+                    function,
+                )? {
+                    ExpressionPointer::Ready { pointer_id } => {
+                        let id = self.id_gen.next();
+                        block
+                            .body
+                            .push(Instruction::load(result_type_id, id, pointer_id, None));
+                        id
+                    }
+                    ExpressionPointer::Conditional { condition, access } => {
+                        let header_block = block.label_id;
```

Comment:
similarly, this code should be a candidate for moving out into a helper

Reply:
I took a shot at this, with, I think, mixed results. Let me know what you think of the current version.